### PR TITLE
[Notifier] Introduce LengthException

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | ---
| License       | MIT
| Doc PR        | ---

Follows https://github.com/symfony/symfony/pull/39444/files#r542298086

### CODE REVIEW

**Review by jderusse — COMMENTED** [review]
I don't see the benefit over an `\InvalidArgumentException`. :/

IMHO not all checks require a dedicated exception class.

File: src/Symfony/Component/Notifier/Exception/LengthException.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * @author Oskar Stark <oskarstark@googlemail.com>
+ *
+ * @experimental in 5.3
+ */
+class LengthException extends LogicException
+{
+    public function __construct(string $message)
+    {
+        parent::__construct($message);
+    }
```
Comment (derrabus) [inline src/Symfony/Component/Notifier/Exception/LengthException.php]:
```suggestion
```

**Comment by derrabus** [comment]
I tend to agree with @jderusse. Can you elaborate why we need a dedicated exception here?

**Comment by OskarStark** [comment]
It was proposed by @chalasr and confirmed by @nicolas-grekas and I felt it makes sense... so... 🤔

**Comment by chalasr** [comment]
Specific exceptions allow for more fine-grained error handling. But both are fine here, I don't really care tbh :)

**Comment by ogizanagi** [comment]
Specific exceptions for logic exceptions don't really make sense to me, appart from normalizing error messages across the codebase.
Actually, I think most of the LogicException in components aren't useful when it doesn't perform any common logic to format the message, nor brings more semantics in the class name, if no native PHP exception matches . I guess we could just use native `\LengthException` here.

**Comment by stof** [comment]
@chalasr but invalid DSNs are not things that you should handle by catching exceptions IMO. You should instead fix your config to provide a valid DSN.

**Comment by OskarStark** [comment]
@stof its not about an invalid DSN in this case, its about a dynamic subject length of a message

**Comment by OskarStark** [comment]
> Specific exceptions for logic exceptions don't really make sense to me, appart from normalizing error messages across the codebase.

@ogizanagi please keep in mind, that new bridges will come from different people and I want to make it more and more easier to implement new ones.

**Review by nicolas-grekas — APPROVED** [review]
Works for me.

**Review by chalasr — APPROVED** [review]

**Comment by ogizanagi** [comment]
Thank you @OskarStark.

**Pull request merged**